### DISCUSSION TITLE
Expected behavior for AutoUpdatePartitions

### DISCUSSION THREAD

**Discussion by ppilev** [fsharplang-ru/pulsar-client-dotnet#290] [General]
hello,
I have a question about how AutoUpdatePartitions(true) should behave when partitioned topic scales up? 
we use KeyShared subscription and encountered an issue that there are no consumers attached to those new partitions thus message backlog starts building up for those partitions.
is there anything else in terms of either server or client configuration that should be considered in such scenarios?

ps: as a workaround atm we use TopicPatterns method

regards,

**Comment by Lanayx** [comment]
Hi! 
This setting is on by default and consumers for new partitions should be added once new partitions are detected. If something doesn't work as you expected, you need to collect logs in debug mode, to understand what's going on. 

Here is a relevant piece of code to look at:
https://github.com/fsprojects/pulsar-client-dotnet/blob/97f012871a2a369fba04d533766577ca3087d413/src/Pulsar.Client/Internal/MultiTopicsConsumerImpl.fs#L439-L522
